app/controllers/locations_controller.rb

```diff
@@ -1,15 +1,35 @@
 class LocationsController < ApplicationController
 
   before_action :authenticate_customer!
+  
+  def index
+    @locations = current_customer.current_account.locations
+  end
 
   def new
     @location = Location.new
   end
   
   def create
-    @location = Location.new(location_params.to_h.merge(account_id: current_customer.current_account_id))
+    @location = current_customer.current_account.locations.build(location_params)
+    if @location.save
+      redirect_to locations_path
+    else
+      respond_to do |format|
+        format.js { render partial: "locations/form", locals: { location: @location }, replace: ".location-form-container" }
+      end
+    end
+  end
+  
+  def edit
+    @location = current_customer.current_account.locations.find(params[:id])
+  end
+  
+  def update
+    @location = current_customer.current_account.locations.build(id: params[:id])
```

Comment:
It updates the record, when Id is present
Below is the Farady request log
I, [2017-03-27T17:13:36.600574 #14970]  INFO -- : put https://stapp.sendaguy.com/api/customers/accounts/238/locations/411